data_steward/gcloud/gcs/__init__.py

```diff
@@ -72,18 +71,17 @@ def get_hpo_bucket(self, hpo_id: str) -> str:
         :return: name of the bucket
         """
         # TODO reconsider how to map bucket name
-        bucket_env = 'BUCKET_NAME_' + hpo_id.upper()
-        hpo_bucket_name = os.getenv(bucket_env)
+        bucket_name: str = self._get_hpo_bucket_id(hpo_site)
 
         # App engine converts an env var set but left empty to be the string 'None'
-        if not hpo_bucket_name or hpo_bucket_name.lower() == 'none':
+        if not bucket_name or bucket_name.lower() == 'none':
             # should not use hpo_id in message if sent to end user.  If the
             # error is logged as a WARNING or higher, this will trigger a
             # GCP alert.
             raise BucketDoesNotExistError(
-                f"Failed to fetch bucket '{hpo_bucket_name}' for hpo_id '{hpo_id}'",
-                hpo_bucket_name)
-        return hpo_bucket_name
+                f"Failed to fetch bucket '{bucket_name}' for hpo '{hpo_site}'",
+                bucket_name)
```

Comment:
> Pretty sure you don't need the `bucket_name` on line 83. Could you just use:

The custom class requires it 😕:

```
class BucketDoesNotExistError(RuntimeError):
    """Raised when a configured bucket does not exist"""

    def __init__(self, msg, bucket):
        super(BucketDoesNotExistError, self).__init__(msg)
        self.bucket = bucket
```

I'm not sure the custom exception actually applies anymore to be honest.  There might be a better way to handle nonexistent buckets through GCS.  Since the exception is raised when there are empty strings, maybe we could raise a GCS exception or simply check using GCS if possible?

Reply:
I don't this any modification is needed here since the exception is useful, and the parameter is needed.